icicle/backend/cpu/include/hash/blake2/blake2s.h

```diff
@@ -0,0 +1,116 @@
+
+#pragma once
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+#include "icicle/hash.h"
+
+using namespace icicle;
+namespace blake2s_cpu {
+
+  typedef unsigned char BYTE;
+  typedef unsigned int WORD;
+  typedef unsigned long long LONG;
+
+#if defined(_MSC_VER)
+#define BLAKE2_PACKED(x) __pragma(pack(push, 1)) x __pragma(pack(pop))
+#else
+#define BLAKE2_PACKED(x) x __attribute__((packed))
+#endif
+
+#if !defined(__cplusplus) && (!defined(__STDC_VERSION__) || __STDC_VERSION__ < 199901L)
+#if defined(_MSC_VER)
+#define BLAKE2_INLINE __inline
+#elif defined(__GNUC__)
+#define BLAKE2_INLINE __inline__
+#else
+#define BLAKE2_INLINE
+#endif
+#else
+#define BLAKE2_INLINE inline
+#endif
```

Comment:
This looks very foolproof :) but all of our modifiers are defined inside utils/modifiers.h. So I think you should either remove it or move to modifiers.h, or use one of already defined ones